src/main/java/org/aksw/iguana/commons/io/ByteArrayListInputStream.java

```diff
@@ -0,0 +1,158 @@
+package org.aksw.iguana.commons.io;
+
+import java.io.EOFException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Objects;
+
+public class ByteArrayListInputStream extends InputStream {
+
+    final List<byte[]> data;
+    Iterator<byte[]> iterator;
+    ByteBuffer currentBuffer;
+    Long available;
+    boolean closed = false;
+
+
+    ByteArrayListInputStream(List<byte[]> data) {
+        this.data = data;
+        this.iterator = data.iterator();
+        if (iterator.hasNext()) {
+            this.currentBuffer = ByteBuffer.wrap(iterator.next());
+        } else {
+            this.currentBuffer = null;
+        }
+    }
+
+    private void updateAvailable(long n) {
+        if (available != null) {
```

Comment:
should this have a check for n>available?

Reply:
I think removing the variable and the method would be better. The available method isn't called that often, so recalculating the available bytes shouldn't be bad.

Reply:
also fine with me.